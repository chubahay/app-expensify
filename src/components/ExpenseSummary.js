import React from 'react'
import getTotalExpenses from '../selectors/expenses-total.js'
import selectExpenses from '../selectors/expenses'
import { connect} from 'react-redux'
import numeral from 'numeral' 

export const ExpenseSummary = ({ numberOfExpenses, expensesTotal }) => (
    <div> 
    <p> Total number of expenses: {numberOfExpenses}</p>
    <p> Total amount of expenses: {numeral(expensesTotal / 100).format('$0,0.00')} </p>
    </div>
)


const mapStateToProps = (state) => {

    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        numberOfExpenses: visibleExpenses.length,
        expensesTotal: getTotalExpenses(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);

