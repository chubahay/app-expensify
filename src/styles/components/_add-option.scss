//add option error

.add-option-error {
    color: $off-white;
    font-style: italic;
    padding: 0 $m-size;
    margin: $m-size 0 0 0;
}

//add option

.add-option {
    display: flex;
    flex-direction: column;
    padding: $m-size;
    
    
}

.button--add-option {
    margin-right: $s-size;
    margin-bottom: 10px;
}

//add option input

.add-option__input {
    background: $dark-blue;
    border: none;
    border-bottom: .3rem solid darken($dark-blue, 10%);
    color: $off-white;
    flex-grow: 1;
    margin-right: $s-size;
    margin-bottom: 10px;
    padding: $s-size;
}

@media (min-width: $desktop-breakpoint) {
    .add-option {
        flex-direction: row;
        justify-content: space-evenly;
    }

    .add-option__input {
        margin-top: 0;
    }


}