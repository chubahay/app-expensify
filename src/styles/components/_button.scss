.button {
    background-color: $blue;
    padding: $s-size;
    font-size: $font-size-large;
    color: white;
    border: none;
}