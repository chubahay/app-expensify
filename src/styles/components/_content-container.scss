.content-container {
    margin: 0;
    padding: 0 $m-size;
    max-width: 80rem;
}