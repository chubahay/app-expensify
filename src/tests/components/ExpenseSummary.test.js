import React from 'react'
import { shallow } from 'enzyme'
import {ExpenseSummary} from '../../components/ExpenseSummary'

test('Should render the ExpenseSummary component correctly with no visible expenses', () => {
    const wrapper = shallow(<ExpenseSummary numberOfExpenses={0} expensesTotal={0}/>)
    expect(wrapper).toMatchSnapshot();
})

test('Should render the correct total number of expenses and total amount of expenses with one visible expense', () => {
    const wrapper = shallow(<ExpenseSummary numberOfExpenses={1} expensesTotal={200}/>)
    expect(wrapper).toMatchSnapshot();  
})

test('Should render the correct total number of expenses and total amount of expenses with two visible expenses', () => {
    const wrapper = shallow(<ExpenseSummary numberOfExpenses={2} expensesTotal={400059}/>)
    expect(wrapper).toMatchSnapshot();  
})
