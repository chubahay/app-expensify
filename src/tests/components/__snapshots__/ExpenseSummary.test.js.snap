// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the ExpenseSummary component correctly with no visible expenses 1`] = `
<div>
  <p>
     Total number of expenses: 
    0
  </p>
  <p>
     Total amount of expenses: 
    $0.00
     
  </p>
</div>
`;

exports[`Should render the correct total number of expenses and total amount of expenses with one visible expense 1`] = `
<div>
  <p>
     Total number of expenses: 
    1
  </p>
  <p>
     Total amount of expenses: 
    $2.00
     
  </p>
</div>
`;

exports[`Should render the correct total number of expenses and total amount of expenses with two visible expenses 1`] = `
<div>
  <p>
     Total number of expenses: 
    2
  </p>
  <p>
     Total amount of expenses: 
    $4,000.59
     
  </p>
</div>
`;
